src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -175,6 +175,13 @@ private function getCurlCommand(array $trace): ?string
         $url = $trace['url'];
         $command = ['curl', '--compressed'];
 
+        if (isset($trace['options']['resolve'])) {
+            $port = parse_url($url, \PHP_URL_PORT) ?: (str_starts_with('http:', $url) ? 80 : 443);
+            foreach ($trace['options']['resolve'] as $host => $ip) {
+                $command[] = '--resolve '.escapeshellarg("$host:$port:$ip");
```

Comment:
You should skip when `$ip` is null. That's an undocumented feature.
https://github.com/symfony/symfony/blob/8e8207bb72d7f2cb8be355994ad2fcfa97c00f74/src/Symfony/Component/HttpClient/CurlHttpClient.php#L178-L180

Reply:
@GromNaN Done https://github.com/symfony/symfony/compare/d501e5778b2937175bf983008a0fe8c7b770b559..88b1b8fd134fa2aa90d6b16517e112dd5bccffae